Refactor pass to class

eellison opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #43743 Update aliasing in tensorexpr fuser
* #43742 use types in the IR instead of vmap
* #43636 Add inline call to profiling executor pipeline
* #43635 Use prim::TensorExprGroup interned symbol
* #43634 Update requires grad property
* #43633 Specialize optionals for grad_sum_to_size
* #43632 Add undefined specializations in backward
* #43631 refactor tests
* **#43630 Refactor pass to class**
* #43629 Add API for unexecuted op

No functional changes here - just refactoring specialize autograd zero to a class, and standardizing its API to take in a shared_ptr<Graph>

Differential Revision: [D23358805](https://our.internmc.facebook.com/intern/diff/D23358805)

Commit:
Refactor pass to class

[ghstack-poisoned]

Commit:
Update on "Refactor pass to class"


No functional changes here - just refactoring specialize autograd zero to a class, and standardizing its API to take in a shared_ptr<Graph>


[ghstack-poisoned]

ZolotukhinM reviewed (approved):
Approving a code move.

Commit:
Update on "Refactor pass to class"


No functional changes here - just refactoring specialize autograd zero to a class, and standardizing its API to take in a shared_ptr<Graph>

Differential Revision: [D23358805](https://our.internmc.facebook.com/intern/diff/D23358805)

[ghstack-poisoned]

Commit:
Update on "Refactor pass to class"


No functional changes here - just refactoring specialize autograd zero to a class, and standardizing its API to take in a shared_ptr<Graph>

Differential Revision: [D23358805](https://our.internmc.facebook.com/intern/diff/D23358805)

[ghstack-poisoned]
